test/no-assign-parameter.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * @fileoverview Tests for no-assign-parameter rule
+ * @author Simon Hajjar <simon.j.hajjar@gmail.com>
+ */
+
+'use strict';
+
+var Solium = require('solium');
+
+var config = {
+        "rules": {
+                "no-assign-parameter": "error"
+        }
+};
+
+describe('[RULE] no-assign-parameter: Acceptances', function () {
+        it('shouldn\'t raise an error for a non-assigned parameter', function(done) {
+                var code = 'contract Blah { function abc(uint256 a, uint256 b) { } }',
+                        errors = Solium.lint(code, config);
+
+                errors.constructor.name.should.equal ('Array');
+                errors.should.be.size(0);
+
+                Solium.reset();
+                done();
+        });
+
+        it('shouldn\'t raise an error for an assigned local variable', function(done) {
```

Comment:
please add tests to see whether it also catches assignments that are inside `if..else` statements and all the loops.

Reply:
Also add tests for incrementing & decrementing integers

Reply:
Good call! Added tests for those cases.